scripts/ci/job_hooks/before_install.sh

```diff
@@ -63,31 +62,37 @@ if [ "${TRAVIS_EVENT_TYPE}" == "push" ]; then
 elif [ "${TRAVIS_EVENT_TYPE}" == "pull_request" ]; then
     echo "pull_request"
     export BASE_HASH="origin/$TRAVIS_BRANCH"
-    source $PARENT_DIR/partials/_ci-flags-parser.sh
+    source "$PARENT_DIR/partials/_ci-flags-parser.sh"
 elif [ "${TRAVIS_EVENT_TYPE}" == "cron" ]; then
     echo "cron"
 else
     echo "api"
 fi
 
 # Cache for protractor smart-runner
-export S3_SMART_RUNNER_PATH="$S3_DBP_PATH/smart-runner/$TRAVIS_BUILD_ID"
+export S3_SMART_RUNNER_PATH="$S3_DBP_PATH/smart-runner/$BUILD_ID"
 
 # Cache for node_modules
-export NODE_VERSION=`node -v`
-export PACKAGE_LOCK_SHASUM=`shasum ./package-lock.json | cut -f 1 -d " "`
+export NODE_VERSION=$(node -v)
+export PACKAGE_LOCK_SHASUM=$(shasum ./package-lock.json | cut -f 1 -d " ")
 # This can change regardless of package-lock.json, so we need to calculate with this one as well
-export S3_NODE_MODULES_CACHE_ID=`echo $NODE_VERSION-$PACKAGE_LOCK_SHASUM | shasum  | cut -f 1 -d " "`
+export S3_NODE_MODULES_CACHE_ID=$(echo "$NODE_VERSION-$PACKAGE_LOCK_SHASUM" | shasum  | cut -f 1 -d " ")
 export S3_NODE_MODULES_CACHE_PATH="$S3_DBP_PATH/cache/node_modules/$S3_NODE_MODULES_CACHE_ID.tar.bz2"
 
 echo "========== Caching settings =========="
-echo "S3_SMART_RUNNER_PATH: $S3_SMART_RUNNER_PATH"
-echo "PACKAGE_LOCK_SHASUM: $PACKAGE_LOCK_SHASUM"
-echo "NODE_VERSION: $NODE_VERSION"
-echo "S3_NODE_MODULES_CACHE_ID: $S3_NODE_MODULES_CACHE_ID"
-echo "S3_NODE_MODULES_CACHE_PATH: $S3_NODE_MODULES_CACHE_PATH"
+echo "S3_SMART_RUNNER_PATH='$S3_SMART_RUNNER_PATH'" | tee -a $ENV_FILE
+echo "PACKAGE_LOCK_SHASUM='$PACKAGE_LOCK_SHASUM'" | tee -a $ENV_FILE
+echo "NODE_VERSION='$NODE_VERSION'" | tee -a $ENV_FILE
+echo "S3_NODE_MODULES_CACHE_ID='$S3_NODE_MODULES_CACHE_ID'" | tee -a $ENV_FILE
+echo "S3_NODE_MODULES_CACHE_PATH='$S3_NODE_MODULES_CACHE_PATH'" | tee -a $ENV_FILE
 echo "========== Nx settings =========="
-echo "GIT_HASH: $GIT_HASH"
-echo "BASE_HASH: $BASE_HASH"
-echo "HEAD_HASH: $HEAD_HASH"
+echo "GIT_HASH='$GIT_HASH'" | tee -a $ENV_FILE
+echo "BASE_HASH='$BASE_HASH'" | tee -a $ENV_FILE
+echo "HEAD_HASH='$HEAD_HASH'" | tee -a $ENV_FILE
 
+
+echo "========== Build vars=========="
+echo "BUILD_OPTS='$BUILD_OPTS'" | tee -a $ENV_FILE
+echo "NX_CALCULATION_FLAGS='$NX_CALCULATION_FLAGS'" | tee -a $ENV_FILE
+echo "HEAD_COMMIT_HASH='$HEAD_COMMIT_HASH'" | tee -a $ENV_FILE
+echo "COMMIT_MESSAGE=\"$COMMIT_MESSAGE\"" | tee -a $ENV_FILE
```

Comment:
I am not sure why are you redirecting to file and echoing to stdout at the same time with tee when you could just run the script and redirect its output to a file.

What am I missing?

keep in mind that in GHA you can easily check which env var has been set expanding each step in the gui

Reply:
ok but make sure with maintainers that they really need this output as it is because it's really duplicating a standard GHA feature

Reply:
i'll append a new task into the epic: as the travis ci is deprecated, i'll use the native way to load the env vars and remove the unused outputs